### PR TITLE
Fix hpopt monitor mode for val_loss tracking metric

### LINKED ISSUE #1287: [v2 BUG]: hpopt uses metric instead of criterion to determine monitor mode for val_loss?
I think we have a bug in our hyperparameter optimization CLI. If the tracking metric is "val_loss", then the monitor mode for should always be "min". But right now it looks at the first metric in the metric list to determine the monitor mode. I think this is a hold over from v2.0 when the "val_loss" was calculated using the first metric in the metric list. I changed this in v2.1 to have the "val_loss" use the same function as "train_loss". In the train CLI if the tracking metric is "val_loss" then the monitor mode is determined based on the criterion class. This wasn't ported to the hpopt CLI. I think the problem would manifest if someone used `--metrics R2` in hpopt, but I haven't confirmed this. Just opening a bug report for now before I can dig into this further. 

Links:
https://github.com/chemprop/chemprop/blob/60befed362b2eae15203a66a7e3791e6e1d9bc3b/chemprop/cli/train.py#L1645
https://github.com/chemprop/chemprop/blob/60befed362b2eae15203a66a7e3791e6e1d9bc3b/chemprop/cli/hpopt.py#L610
https://github.com/chemprop/chemprop/blob/73d8948f05c044e671562412fa1c295cf5d05e45/chemprop/models/model.py#L161
https://github.com/chemprop/chemprop/blob/60befed362b2eae15203a66a7e3791e6e1d9bc3b/chemprop/models/model.py#L181

### PR DESCRIPTION
**Pull request by JacksonBurns**
> [!NOTE]
> This PR was handled entirely by Qwen 3.6 27b running locally via `opencode` with @JacksonBurns' supervision

Use the criterion class's higher_is_better instead of the first metric's to determine monitor mode when tracking val_loss, matching the logic in train.py. Previously, using --metrics R2 would incorrectly set monitor_mode to 'max' for val_loss.

Resolves #1287 

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by JacksonBurns** [comment]
@KnathanM I think this resolves the linked issue - could you take a look?

**Review by KnathanM — APPROVED** [review]
